src/Avalonia.Controls/Primitives/PopupRoot.cs

```diff
@@ -74,7 +74,7 @@ public PopupRoot(TopLevel parent, IPopupImpl impl, IAvaloniaDependencyResolver d
         IStyleHost IStyleHost.StylingParent => Parent;
 
         /// <inheritdoc/>
-        public void Dispose() => PlatformImpl?.Dispose();
+        public void Dispose() => HandleClosed();
```

Comment:
Huh, you know what, it doesn't. I saw it was setting it to `null` but it doesn't first dispose it, i guess because a `Close`ing the platform impl is the same as disposing it? Looks like we need to dispose the platform impl and the call `HandleClosed` to set the pointer to null?

It's all a bit confusing...

Reply:
Ok, `Window` disposes the `PlatformImpl` in `CloseInternal` which is called via `HandleClosing`, i.e. before `HandleClosed`, so I've made `PopupRoot` dispose the `PlatformImpl` before calling `HandleClosed` -- does that look right now?

Reply:
IIRC it was supposed to work like this:
1) Disposing WindowBase would close the window and will trigger the usual close sequence but without `Closing` and cancellation
2) During such procedure the platform impl would call `Closed` callback

So `HandleClosed` should be called by platform impl itself